src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1422,7 +1422,7 @@ public bool DoIndexingWork(IndexingStatsScope stats, CancellationToken cancellat
                 using (CurrentIndexingScope.Current =
                     new CurrentIndexingScope(DocumentDatabase.DocumentsStorage, databaseContext, Definition, indexContext, GetOrAddSpatialField))
                 {
-                    var writeOperation = new Lazy<IndexWriteOperation>(() => IndexPersistence.OpenIndexWriter(indexContext.Transaction.InnerTransaction));
+                    var writeOperation = new Lazy<IndexWriteOperation>(() => IndexPersistence.OpenIndexWriter(indexContext.Transaction.InnerTransaction, indexContext));
```

Comment:
Isn't the indexContext going to be disposed soon? Can the lazy be evaluated _after_ the context has been disposed?